### DISCUSSION TITLE
Option to disable Tripwire block-state update

### DISCUSSION THREAD

**Discussion by Boy0000** [PurpurMC/Purpur#937] [Feature Requests]
Similar to how mushroom-blocks and noteblocks have this functionality, it would be great for tripwires.
Very useful for making custom-block plugins, but tripwires can only work if it is baked into the server.

Great for adding decoration and nature-stuff as they will be passthroughable etc

https://user-images.githubusercontent.com/62521371/159251528-3e1a472e-3c54-4a7f-9c5d-94c3df91cfbd.mp4

**Comment by IllagerCaptain** [comment]
Could really use this.

**Reply by Boy0000** [reply]
Seems this can be done by simply checking EntityInsideEvent
[Used it here](https://github.com/MineInAbyss/Blocky/blob/master/src/main/kotlin/com/mineinabyss/blocky/listeners/BlockyTripwireListener.kt#L59-L61)
```kotlin
@EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
   fun EntityInsideBlockEvent.onEnterTripwire() {
        if (block.type == Material.TRIPWIRE) isCancelled = true
    }
